webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/repository/xml/GitArchiverUtils.scala

```diff
@@ -200,14 +206,29 @@ trait GitArchiverUtils {
     )
   }
 
-  def toGitPath(fsPath:File) = fsPath.getPath.replace(gitRootDirectory.getPath +"/","")
+  def toGitPath(fsPath: File) = fsPath.getPath.replace(gitRootDirectory.getPath +"/","")
 
   /**
-   * Write the given Elem (prettified) into given file, log the message
+   * Write the given Elem (prettified) into given file, log the message.
+   * File are written with the following rights:
+   * - for directories, rwxrwx---
+   * - for files: rw-rw----
+   * - for all, group owner: rudder
+   * Perms are defined in /opt/rudder/bin/rudder-fix-repository-configuration
    */
-  def writeXml(fileName:File, elem:Elem, logMessage:String) : IOResult[File] = {
+  def writeXml(fileName: File, elem: Elem, logMessage: String) : IOResult[File] = {
+    import java.nio.file.attribute.PosixFilePermission._
+    import java.nio.file.StandardOpenOption._
+    import better.files._
+    val filePerms = Set[PosixFilePermission](OWNER_READ, OWNER_WRITE, /* no exec, */ GROUP_READ, GROUP_WRITE /* no exec, */ )
+    val directoryPerms = Set[PosixFilePermission](OWNER_READ, OWNER_WRITE, OWNER_EXECUTE, GROUP_READ, GROUP_WRITE, GROUP_EXECUTE)
+
+    // an utility that write text in a file and create file parents if needed
+    // open file mode for create or overwrite mode
     IOResult.effect {
-      FileUtils.writeStringToFile(fileName, xmlPrettyPrinter.format(elem), encoding)
+      val file = fileName.toScala
+      file.parent.createDirectoryIfNotExists(true).setPermissions(directoryPerms).setGroup(groupOwner)
```

Comment:
Does the directory created is the same as `fileName` ?

Reply:
I don't understand the question. Can you explain a bit more?